Dockerfile

```
@@ -21,7 +21,8 @@ FROM base as production-deps
 WORKDIR /myapp
 
 COPY --from=deps /myapp/node_modules /myapp/node_modules
-ADD package.json package-lock.json ./
+# A wildcard is used to ensure both package.json AND package-lock.json are copied
+ADD package*.json ./
```

`package.json` is already added, so shouldn't this just add `package-lock.json` here? 🤔

---

we could do something like this too https://stackoverflow.com/a/70096420/6626541

---

If replacing it with the following works, I think that would be a better solution than replacing a placeholder in `remix.init`

```suggestion
ADD [package-lock.json|pnp-lock.y?(a)ml|yarn.lock] ./
```

CC/ @kentcdodds What do you think about this?

---

I'll verify real quick locally, but yeah that's essentially what every answer is for "copy a file only if it exists" comes up with for Docker

---

hm, we also run an `npm install` for them, so there might be a bit more to do

---

If it's not possible to do so, I would suggest to just keep the `npm install` for now

Once `v1.5.0` is released, we'll have `packageManager` available in `remix.init`, so we can do some crazy stuff with it if we want to.

---

Yeah, since this issue isn't impacting people, let's go ahead and wait for 1.5.0 to be released so we can have a good final solution.

---

1.5.0 was released, we can resume work here :)

---

decided to spent a few minutes to make this work via a quick and easy bash script 8ce6113